website/static/js/file-browser.js

```diff
@@ -0,0 +1,284 @@
+                self.isLoadedUrl = true;
+            }
+        }.bind(self);
+
+        // BREADCRUMBS
+        self.updateBreadcrumbs = function(linkObject){
+            var crumb = {
+                label : linkObject.label,
+                url : linkObject.link,
+                id : linkObject.id
+            };
+            if (linkObject.type === 'collection' || linkObject.type === 'user'){
+                self.breadcrumbs([crumb]);
+                return;
+            }
+            if (linkObject.type === 'breadcrumb'){
+                self.breadcrumbs().splice(linkObject.index+1, self.breadcrumbs().length-linkObject.index-1);
+                return;
+            }
+            self.breadcrumbs().push(crumb);
+        }.bind(self);
+
+        self.generateLinks = function (linkObject) {
+            if (linkObject.type === 'collection'){
+                return $osf.apiV2Url(linkObject.data.path, linkObject.data.pathOptions);
+            }
+            if (linkObject.type === 'breadcrumb') {
+                return linkObject.data.url;
+            }
+            if (linkObject.type === 'user') {
+                return $osf.apiV2Url('users/' + linkObject.id + '/nodes', {});
+            }
+            return $osf.apiV2Url('nodes/' + linkObject.data.uid + '/children', {}); // If type === node
+        };
+
+
+
+    },
+    view : function (ctrl) {
+        var infoPanel = '';
+        var poStyle = 'width : 75%';
+        var infoClass = 'btn-default';
+        if(ctrl.showInfo()){
+            infoPanel = m('.fb-infobar', m.component(Information, { selected : ctrl.selected }));
+            poStyle = 'width : 55%';
+            infoClass = 'btn-primary';
+        }
+
+        return m('', [
+            m('.fb-header', [
+                m.component(Breadcrumbs, { data : ctrl.breadcrumbs, updateFilesData : ctrl.updateFilesData}),
+                m('.fb-buttonRow', [
+                    m('button.btn', {
+                        'class' : infoClass,
+                        onclick : function () {
+                            ctrl.showInfo(!ctrl.showInfo());
+                        }
+                    }, m('.fa.fa-info'))
+                ])
+            ]),
+            m('.fb-sidebar', [
+                m.component(Collections, {list : ctrl.collections, activeCollection : ctrl.activeCollection, updateCollection : ctrl.updateCollection } ),
+                m.component(Filters, { activeUser : ctrl.activeUser, updateUser : ctrl.updateUserFilter, nameFilters : ctrl.nameFilters, tagFilters : ctrl.tagFilters })
+            ]),
+            m('.fb-main', { config: ctrl.updateList, style : poStyle },
+                m('#poOrganizer', m('.spinner-loading-wrapper', m('.logo-spin.logo-md')))
+            ),
+            infoPanel
+        ]);
+    }
+};
+
+/**
+ * Collections Module.
+ * @constructor
+ */
+var Collections  = {
+    controller : function (args) {
+        var self = this;
```

Comment:
I'm struggling with this, it's something mithril uses and it's supposed to be super general.